### ISSUE TITLE
Expose activeStreams and potentially pendingStreams for use in ConnectionPoolMetrics in MeterRegistrar

### ISSUE THREAD

**Issue by FlattestWhite** [reactor/reactor-netty#3344]
We use our own metrics vendor and don't use micrometer. To instrument the httpclient, we have an implementation of MeterRegistrar. There's currently no way to get access to activeStreams in ConnectionPoolMetrics. The provided micrometer metrics internally cast to an `Http2Pool` before calling its activeStreams method.

## Motivation

To work around this, we have to keep a count of concurrent requests through a counter (increment and decrement) and report that. Would prefer to have it come through in ConnectionPoolMetrics

**Comment by violetagg** [comment]
@FlattestWhite Would you like to create a PR? You didn't mention Reactor Netty version.

**Comment by violetagg** [comment]
> Just wanted to confirm that activeStreams metric isn't available through existing interface. If that's the case, I'll look to create a PR sometime for it.

Yeah it is not exposed there as this is `reactor-core` and we cannot add `http` related information. I think that we can provide something in `reactor-netty` that extends `ConnectionPoolMetrics` and adds this information.

**Comment by raccoonback** [comment]
@violetagg 

Hello!
It seems that the issue has not been progressed, so I would like to ask if it is okay for me to handle it.

Could you confirm if my understanding of the issue is correct?
1. Add metrics for `activeStreams` and `pendingStreams` in the HTTP/2 protocol.
2. Collect and expose HTTP/2 metrics via `DelegatingConnectionPoolMetrics` or `Micrometer`.

Also, I was wondering if there would be any additional data to aggregate besides `activeStreams` and `pendingStreams`.
If there's anything I might have missed or misunderstood, I’d appreciate your feedback.

Thanks!

**Comment by violetagg** [comment]
> [@violetagg](https://github.com/violetagg)
> 
> Hello! It seems that the issue has not been progressed, so I would like to ask if it is okay for me to handle it.
> 
> Could you confirm if my understanding of the issue is correct?
> 
> 1. Add metrics for `activeStreams` and `pendingStreams` in the HTTP/2 protocol.
> 2. Collect and expose HTTP/2 metrics via `DelegatingConnectionPoolMetrics` or `Micrometer`.
> 
> Also, I was wondering if there would be any additional data to aggregate besides `activeStreams` and `pendingStreams`. If there's anything I might have missed or misunderstood, I’d appreciate your feedback.
> 
> Thanks!

Yes this is correct and I don't think we need more data to be exposed. One additional thing is let's expose these data also for HTTP/3
